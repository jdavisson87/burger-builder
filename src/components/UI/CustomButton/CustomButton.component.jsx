import React from 'react';
import { CustomButtonContainer } from './CustomButton.styles';

const CustomButton = ({ btnType, clicked, children }) => (
  <CustomButtonContainer
    data-testid="CustomButton"
    btnType={btnType}
    onClick={clicked}
  >
    {children}
  </CustomButtonContainer>
);

export default CustomButton;
