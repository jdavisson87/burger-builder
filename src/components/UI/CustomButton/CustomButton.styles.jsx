import styled from 'styled-components';

const colors = {
  success: '#5C9210',
  danger: '#944317',
  blue: '#0000FF',
};

export const CustomButtonContainer = styled.button`
  background-color: transparent;
  border: none;
  color: white;
  outline: none;
  cursor: pointer;
  font: inherit;
  padding: 10px;
  margin: 10px;
  font-weight: bold;

  &:first-of-type {
    margin-left: 0;
    padding-left: 0;
  }

  color: ${(props) => colors[props.btnType]};
`;
