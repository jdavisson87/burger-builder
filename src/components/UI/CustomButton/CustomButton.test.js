import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import CustomButton from './CustomButton.component';
import '@testing-library/jest-dom/extend-expect';

afterEach(() => {
  cleanup();
  onClickMock.mockClear();
});
const onClickMock = jest.fn();

test('fires when the button is clicked', () => {
  const { getByTestId } = render(
    <CustomButton onClick={onClickMock()}>Testing</CustomButton>
  );
  expect(getByTestId('CustomButton')).toBeTruthy();
  expect(getByTestId('CustomButton')).toHaveTextContent('Testing');

  fireEvent.click(getByTestId('CustomButton'));

  expect(onClickMock).toHaveBeenCalledTimes(1);
});

test('snapshot', () => {
  const { container } = render(
    <CustomButton onClick={onClickMock()} btnType="green">
      Testing
    </CustomButton>
  );
  expect(container.firstChild).toMatchSnapshot();
});
