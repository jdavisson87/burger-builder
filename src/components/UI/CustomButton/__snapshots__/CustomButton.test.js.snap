// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<button
  data-testid="CustomButton"
  class="sc-AxjAm iXVEia"
>
  Testing
</button>
`;
